runelite-client/src/main/java/net/runelite/client/plugins/itemstats/potions/MoonlightPotion.java

```diff
@@ -0,0 +1,138 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.itemstats.potions;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import net.runelite.api.Client;
+import net.runelite.api.Skill;
+import net.runelite.client.plugins.itemstats.Effect;
+import net.runelite.client.plugins.itemstats.SimpleStatBoost;
+import net.runelite.client.plugins.itemstats.StatBoost;
+import net.runelite.client.plugins.itemstats.StatChange;
+import net.runelite.client.plugins.itemstats.StatsChanges;
+import net.runelite.client.plugins.itemstats.delta.DeltaCalculator;
+import net.runelite.client.plugins.itemstats.stats.Stat;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.stream.Stream;
+import static net.runelite.client.plugins.itemstats.Builders.perc;
+import static net.runelite.client.plugins.itemstats.stats.Stats.ATTACK;
+import static net.runelite.client.plugins.itemstats.stats.Stats.DEFENCE;
+import static net.runelite.client.plugins.itemstats.stats.Stats.HERBLORE;
+import static net.runelite.client.plugins.itemstats.stats.Stats.PRAYER;
+import static net.runelite.client.plugins.itemstats.stats.Stats.STRENGTH;
+
+@RequiredArgsConstructor
+public class MoonlightPotion implements Effect
+{
+	private static final Stat[] OFFENSIVE_MELEE_STATS = {
+			ATTACK, STRENGTH
+	};
+
+	final StatBoost prayerRestore = new StatBoost(PRAYER, false)
+	{
+		@Override
+		public int heals(Client client)
+		{
+			int pray = PRAYER.getMaximum(client);
+			int herb = HERBLORE.getValue(client);
+			return herb < 38 ? 0 : (int) Math.max(pray * 0.25, herb * 0.3) + 7;
+		}
+	};
+
+	@Override
+	public StatsChanges calculate(Client client)
+	{
+		StatsChanges changes = new StatsChanges(0);
+		StatChange restoreChange = prayerRestore.effect(client);
+
+		SimpleStatBoost attackBoost = new SimpleStatBoost(ATTACK, true, getMeleeEffect(client, ATTACK));
+		SimpleStatBoost defenceBoost = new SimpleStatBoost(DEFENCE, true, getMeleeEffect(client, DEFENCE));
+		SimpleStatBoost strengthBoost = new SimpleStatBoost(STRENGTH, true, getMeleeEffect(client, STRENGTH));
+
+		changes.setStatChanges(Stream.of(
+			Stream.of(restoreChange),
+			Stream.of(defenceBoost.effect(client)),
+			Stream.of(OFFENSIVE_MELEE_STATS)
+				.map(stat ->
+				{
+					if (stat == ATTACK)
+					{
+						attackBoost.setStat(stat);
+						return attackBoost.effect(client);
+					}
+					if (stat == STRENGTH)
+					{
+						strengthBoost.setStat(stat);
+						return strengthBoost.effect(client);
+					}
+					return null;
+				}))
```

Comment:
Why are the prayer restore, defense boost, and attack/strength boosts all assembled in completely different ways here? `restoreChange` is not reused, so the variable can be inlined. Both the attack and strength boosts already have stats set because they're initialized as `SimpleStatBoost`s, so the `setStat()` calls are no-ops. `OFFENSIVE_MELEE_STATS` is similarly not reused, so I'm not really sure what purpose it serves here.

Anyway, the only potential purpose I see to using a Stream here is to exclude the prayer restore if the player's herblore level is less than 38, but even then it's probably fine to list "+0 Prayer" for this boost, in which case this whole stream-to-array process can be replaced by a simple `StatChange[]` var declaration.